### PR TITLE
Unshred-tag API client.

### PR DESCRIPTION
**Pull request by xa4a**
A JS library for interacting with the unshred-tag web API. 
This is intended to be used for stitching UI. 
JS is not my primary language, so any comments are welcome.

[Review on Reviewable](https://reviewable.io/reviews/dchaplinsky/unshred-tag/75)

### CODE REVIEW

File: static/scripts/unshred_api_client.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    // Angle in degrees.
+    //
+    // Translate to cluster1's [0, 0] as origin.
+    position2[0] -= position1[0];
+    position1[0] -= position1[0];
+    position2[1] -= position1[1];
+    position1[1] -= position1[1];
+
+    // Angle to rad.
+    angle1 *= Math.PI / 180;
```
Comment (dchaplinsky) [inline static/scripts/unshred_api_client.js]: Helper?

File: static/scripts/unshred_api_client.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+ * @param error - Failure callback.
+ */
+UnshredAPIClient.prototype.MergeClusters = function (cluster1, position1,
+                                                     angle1, cluster2,
+                                                     position2, angle2,
+                                                     success, error) {
+    // Angle in degrees.
+    //
+    // Translate to cluster1's [0, 0] as origin.
+    position2[0] -= position1[0];
```
Comment (dchaplinsky) [inline static/scripts/unshred_api_client.js]: Helper?

File: static/scripts/unshred_api_client.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    for (var i = 0; i < cluster1.members.length; i++) {
+        var origMember = cluster1.members[i];
+        var member = {
+            shred: origMember.shred,
+            position: origMember.position,
+            // Angle to degrees.
+            angle: origMember.angle
+        };
+
+        members[members.length] = member;
```
Comment (dchaplinsky) [inline static/scripts/unshred_api_client.js]: push?

File: static/scripts/unshred_api_client.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+            members: members,
+        }
+    };
+
+    console.log(data);
+    this.Request("POST", "cluster", function (response_data) {
+        success(response_data.id);
+    }, error, data);
+    return data;
+};
```
Comment (dchaplinsky) [inline static/scripts/unshred_api_client.js]: That function screams for some tests

File: static/scripts/unshred_api_client.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+    // Angle to rad.
+    angle1 *= Math.PI / 180;
+    angle2 *= Math.PI / 180;
+
+    var members = [];
+
+    // Transform cluster1's members.
+    for (var i = 0; i < cluster1.members.length; i++) {
+        var origMember = cluster1.members[i];
+        var member = {
```
Comment (dchaplinsky) [inline static/scripts/unshred_api_client.js]: Helper?

File: static/scripts/unshred_api_client.js (hunk 1 above)
Comment (xa4a) [inline static/scripts/unshred_api_client.js]: Done.

File: static/scripts/unshred_api_client.js (hunk 4 above)
Comment (xa4a) [inline static/scripts/unshred_api_client.js]: Ack.

File: static/scripts/unshred_api_client.js (hunk 5 above)
Comment (xa4a) [inline static/scripts/unshred_api_client.js]: Done.

File: static/scripts/unshred_api_client.js (hunk 3 above)
Comment (xa4a) [inline static/scripts/unshred_api_client.js]: Done.

**Pull request merged**